### DISCUSSION TITLE
How to stop scheduled task?

### DISCUSSION THREAD

**Discussion by goodstas** [jamesmh/coravel#385] [Q&A]
Hi, i want to integrate Coravel nuget in my projet and i want to understand how i stop scheduler.
This is how i schedule new task  :
provider.UseScheduler(scheduler =>
{
    scheduler.Schedule(
        () => Console.WriteLine("Every minute during the week.")
    )
    .EveryMinute()
    .Weekday();
});

But how i can stop it? 
I read the documentation from [here](https://docs.coravel.net/Scheduler/) but i did not find any reference how i stop it.

Another question that i have what is the best way to schedule tasks when i don't know from the application startup what i need to schedule. 
My scenario is as follows first of all the application needs to query some service for the information and after that it will know to schedule tasks based on this information.

Thank you!

**Comment by jamesmh** [comment]
Hi! Check out this issue for details on what you can try: https://github.com/jamesmh/coravel/issues/216

This is not an official feature at the moment, but you can take this approach.

**Reply by goodstas** [reply]
@jamesmh thank you. i will try to implement this approach. 
Are you going to implement some simpler solution to this issue? As i see people are asked for it quite a long time.
Thank you again.

**Comment by jamesmh** [comment]
Related to https://github.com/jamesmh/coravel/issues/141

**Comment by goodstas** [comment]
@jamesmh I thought about another option to stop long running task. I can schedule it to run once and to setup the invocable with cancellation token. Once i set the cancellation token , the invokable will exit and the task should be stopped.
Is is a good approach? Will it work?

**Comment by tesar-tech** [comment]
I checked all the other comments and issues and came up with minimal demonstration of "how to cancel scheduled task":

```cs
//Program.cs
builder.Services.AddScheduler();

var app = builder.Build();

IScheduler scheduler = app.Services.GetService<IScheduler>()!;
int ii = 0;
scheduler.Schedule(()=> {
    Console.WriteLine("Hit the scheduler ");
    if (ii++ >= 3)
    {
        bool isSuccess = (scheduler as Scheduler)!.TryUnschedule("SomeId");
        Console.WriteLine($"Is Success: {isSuccess}");
    }
}).EverySecond().PreventOverlapping("SomeId");
```

Results in:

```sh
Hit the scheduler
Hit the scheduler
Hit the scheduler
Hit the scheduler
Is Success: True
```

**Reply by handirusli** [reply]
> I checked all the other comments and issues and came up with minimal demonstration of "how to cancel scheduled task":
> 
> ```cs
> //Program.cs
> builder.Services.AddScheduler();
> 
> var app = builder.Build();
> 
> IScheduler scheduler = app.Services.GetService<IScheduler>()!;
> int ii = 0;
> scheduler.Schedule(()=> {
>     Console.WriteLine("Hit the scheduler ");
>     if (ii++ >= 3)
>     {
>         bool isSuccess = (scheduler as Scheduler)!.TryUnschedule("SomeId");
>         Console.WriteLine($"Is Success: {isSuccess}");
>     }
> }).EverySecond().PreventOverlapping("SomeId");
> ```
> 
> Results in:
> 
> ```shell
> Hit the scheduler
> Hit the scheduler
> Hit the scheduler
> Hit the scheduler
> Is Success: True
> ```

Hi @tesar-tech Is it okay if I use the same logic as you described for a new Job ?
